### ISSUE TITLE
TrxReport Capability Documentation

### ISSUE THREAD

**Issue by thomhurst** [microsoft/testfx#3196]
Hey, is there any documentation on implementing Trx Report capabilities? I've tried to add some properties based on the xmldocs on the  `ITrxReportCapability` interface.

I've added `trxreport.exceptionmessage` and `trxreport.exceptionstacktrace`  properties but I'm not getting any error info in my trx file.

This is how I'm adding it:

```csharp
 await context.MessageBus.PublishAsync(this, new TestNodeUpdateMessage(context.Request.Session.SessionUid, test.TestNode
                .WithProperty(new FailedTestNodeStateProperty(e))
                .WithProperty(new TimingProperty(new TimingInfo(start ?? end, end, start.HasValue ? end-start.Value : TimeSpan.Zero)))
                .WithProperty(new KeyValuePairStringProperty("trxreport.exceptionmessage", e.Message))
                .WithProperty(new KeyValuePairStringProperty("trxreport.exceptionstacktrace", e.StackTrace!))));
```

Where `WithProperty` is a simple extension method:

```csharp
    public static TestNode WithProperty(this TestNode testNode, IProperty property)
    {
        testNode.Properties.Add(property);
        return testNode;
    }
```

**Comment by MarcoRossignoli** [comment]
You're right we should add some documentation here https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-test-reports#visual-studio-test-reports

cc: @Evangelink

**Comment by Evangelink** [comment]
I think this is more of a technical documentation for framework authors, so we should start with doc on our repo.

@MarcoRossignoli we should ask @IEvangelist if we can start moving the technical doc under the official doc repo so we have something unified.

**Comment by IEvangelist** [comment]
Move what, where?

**Comment by Evangelink** [comment]
Sorry :) We started to write some technical documentation (for framework and extensions authors) around our new testing platform. At the moment this tech doc, lives in this repo https://github.com/microsoft/testfx/blob/main/docs/testingplatform/Index.md but I think it'd be better overall if all the doc was available at the same location (with some structure to split "normal user" from "extension builder user").
